src/components/App/App.component.jsx

```diff
@@ -1,55 +1,45 @@
-import React, { useLayoutEffect } from 'react';
+import React from 'react';
 import { BrowserRouter, Switch, Route } from 'react-router-dom';
 
+import { ThemeProvider } from 'styled-components';
+import { GlobalStyles } from '../Theme/globalStyles.component';
 import AuthProvider from '../../providers/Auth';
+import FavoritesProvider from '../../providers/Favorites/Favorites.provider';
+import FavoritePage from '../../pages/Favorites/Favorite.page';
 import HomePage from '../../pages/Home';
-import LoginPage from '../../pages/Login';
+import Navbar from '../Navbar/Navbar.component';
 import NotFound from '../../pages/NotFound';
-import SecretPage from '../../pages/Secret';
+import SearchProvider from '../../providers/Search/Search.provider';
 import Private from '../Private';
-import Fortune from '../Fortune';
 import Layout from '../Layout';
-import { random } from '../../utils/fns';
+import VideoDetail from '../../pages/VideoDetail/VideoDetail.component';
+import { useTheme } from '../../providers/Theme/Theme.provider';
+import { lightTheme, darkTheme } from '../Theme/Theme.component';
 
 function App() {
-  useLayoutEffect(() => {
-    const { body } = document;
-
-    function rotateBackground() {
-      const xPercent = random(100);
-      const yPercent = random(100);
-      body.style.setProperty('--bg-position', `${xPercent}% ${yPercent}%`);
-    }
-
-    const intervalId = setInterval(rotateBackground, 3000);
-    body.addEventListener('click', rotateBackground);
-
-    return () => {
-      clearInterval(intervalId);
-      body.removeEventListener('click', rotateBackground);
-    };
-  }, []);
+  const { theme } = useTheme();
 
   return (
     <BrowserRouter>
       <AuthProvider>
-        <Layout>
-          <Switch>
-            <Route exact path="/">
-              <HomePage />
-            </Route>
-            <Route exact path="/login">
-              <LoginPage />
-            </Route>
-            <Private exact path="/secret">
-              <SecretPage />
-            </Private>
-            <Route path="*">
-              <NotFound />
-            </Route>
-          </Switch>
-          <Fortune />
-        </Layout>
+        <SearchProvider>
+          <FavoritesProvider>
+            <ThemeProvider theme={theme === 'light' ? lightTheme : darkTheme}>
+              <>
+                <GlobalStyles />
+                <Layout>
+                  <Navbar />
+                  <Switch>
+                    <Route exact path="/" component={HomePage} />
```

Comment:
I strongly recommend to separate the logic of your routes in a different component, this way you separate concerns and keep abstract your components :)